gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1268 @@
+        for p in pipes:
+            answer = p[0].recv()  # [0], because that is the parentConn
+            p[0].close()  # child conn will be closed from inside the worker
+            # this does basically the same as the generate_topic_models function (concatenate all the ttdas):
+            distances += [answer[1]]
+
+        # end all processes
+        for p in processes:
+            p.terminate()
+
+        self.asymmetric_distance_matrix = np.concatenate(distances)
+
+        return self.asymmetric_distance_matrix
+
+    def calculate_asymmetric_distance_matrix_chunk(self, ttda1, ttda2, threshold, start_index, method):
+        """Iterates over ttda1 and calculates the
+        distance to every ttda in tttda2.
+
+        Parameters
+        ----------
+        ttda1 and ttda2: 2D arrays of floats
+            Two ttda matrices that are going to be used for distance calculation.
+            Each row in ttda corresponds to one topic. Each cell in the resulting
+            matrix corresponds to the distance between a topic pair.
+        threshold : float, optional
+            threshold defaults to: {"mass": 0.95, "rank": 0.11}, depending on the selected
+            method
+        start_index : number
+            this function might be used in multiprocessing, so start_index has to be set as
+            ttda1 is a chunk of the complete ttda in that case. start_index would be 0
+            if ttda1 == self.ttda. When self.ttda is split into two pieces, each 100 ttdas
+            long, then start_index should be be 100. default is 0
+        method : {'mass', 'rank}, optional
+            method can be "mass" for the original masking method or "rank" for a faster
+            masking method that selects by rank of largest elements in the topic word
+            distribution, to determine which tokens are relevant for the topic.
+
+        Returns
+        -------
+        2D Numpy.numpy.ndarray of floats
+        """
+
+        if method not in ["mass", "rank"]:
+            raise ValueError("method {} unknown".format(method))
+
+        # select masking method:
+        def mass_masking(a):
+            """original masking method. returns a new binary mask"""
+            sorted_a = np.sort(a)[::-1]
+            largest_mass = sorted_a.cumsum() < threshold
+            smallest_valid = sorted_a[largest_mass][-1]
+            return a >= smallest_valid
+
+        def rank_masking(a):
+            """faster masking method. returns a new binary mask"""
+            return a > np.sort(a)[::-1][int(len(a) * threshold)]
+
+        create_mask = {"mass": mass_masking, "rank": rank_masking}[method]
+
+        # some help to find a better threshold by useful log messages
+        avg_mask_size = 0
+
+        # initialize the distance matrix. ndarray is faster than zeros
+        distances = np.ndarray((len(ttda1), len(ttda2)))
+
+        # now iterate over each topic
```

Comment:
Please consider rewriting your code comments to be more helpful. Comments should explain **why** you're doing something, not **what**, because what you are doing should already be obvious from the code.

If you find yourself writing code where it's **not** obvious what you're doing, then instead of adding comments, please consider making the code easier to read by refactoring, using more helpful variable names, extracting code blocks to functions, etc.

Reply:
thanks. done